Segmentation fault while running unit tests on macOS Ventura / arm64 / Xcode 14.3

Tested on current master, 128c937335ce23d35c4e98c3b7618d7420bd11ee.

`ninja` fails while trying to execute the unit tests:
```
CMake Error at /Users/tinow/venv-py311/lib/python3.11/site-packages/cmake/data/share/cmake-3.26/Modules/GoogleTestAddTests.cmake:112 (message):
  Error running test executable.

    Path: '/Users/tinow/zi/git/ibm/qss-compiler/build-release/test/unittest/unittest-qss-compiler'
    Result: Segmentation fault
    Output:
```

Debugger output (release build, but it has some context):
```
❯ lldb ./test/unittest/unittest-qss-compiler
(lldb) target create "./test/unittest/unittest-qss-compiler"
Current executable set to '/Users/tinow/zi/git/ibm/qss-compiler/build-release/test/unittest/unittest-qss-compiler' (arm64).
(lldb) run
Process 5819 launched: '/Users/tinow/zi/git/ibm/qss-compiler/build-release/test/unittest/unittest-qss-compiler' (arm64)
Process 5819 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x00000001022aa6cc unittest-qss-compiler`QASM::ASTDeclarationContextTracker::Init() [inlined] std::__1::__tree<std::__1::__value_type<unsigned int, QASM::ASTDeclarationContext const*>, std::__1::__map_value_compare<unsigned int, std::__1::__value_type<unsigned int, QASM::ASTDeclarationContext const*>, std::__1::less<unsigned int>, true>, std::__1::allocator<std::__1::__value_type<unsigned int, QASM::ASTDeclarationContext const*> > >::__insert_node_at(this=<unavailable>, __parent=0x0000000103108bd8, __child=0x0000000103108bd8, __new_node=0x0000600000c001e0) at __tree:2078:25 [opt]
   2075	    __new_node->__parent_ = __parent;
   2076	    // __new_node->__is_black_ is initialized in __tree_balance_after_insert
   2077	    __child = __new_node;
-> 2078	    if (__begin_node()->__left_ != nullptr)
   2079	        __begin_node() = static_cast<__iter_pointer>(__begin_node()->__left_);
   2080	    _VSTD::__tree_balance_after_insert(__end_node()->__left_, __child);
   2081	    ++size();
Target 0: (unittest-qss-compiler) stopped.
warning: unittest-qss-compiler was compiled with optimization - stepping may behave oddly; variables may not be available.
(lldb)
```

Additional info:
```
❯ conan install .. --profile default --update --build=outdated
Configuration:
[settings]
arch=armv8
arch_build=armv8
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=14
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

...
```

```
❯ clang --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

--- a/conanfile.py
+++ b/conanfile.py
@@ -42,6 +42,10 @@ class QSSCompilerConan(ConanFile):
         for req in self.conan_data["requirements"]:
             self.requires(req)
 
+    def configure(self):
+        if self.settings.os == "Macos":
+            self.options["qasm"].shared = True
+
     def build_requirements(self):
         tool_pkgs = ["llvm", "clang-tools-extra"]
         # Add packages necessary for build.

